apis/destination.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apis
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"net/url"
+	"strings"
+)
+
+// Destination represents a target of an invocation over HTTP.
+type Destination struct {
```

Comment:
I'm wondering if this belongs in a versioned directory...

Reply:
not a bad idea. Moved to apis/v1alpha1 

I am not sure if we want to give it a name or just a apis with version. It felt more natural to have it `apis/v1alpha1`